### PR TITLE
Modify DataServlet.java to take name parameter

### PR DESCRIPTION
**Pull request by keiffer01**
Started the tasks for week 3 solo development. First task was to modify a servlet file. I had it take in a name parameter and output "Hello [name]!"

TESTING: tested various strings of different lengths. Also tested if the name parameter was not specified, with it outputting "Oops, I don't know what your name is." if so. (It's also susceptible to XSS attacks oop, luckily this is still just a tutorial).

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,7 +26,16 @@
...
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, I don't know what your name is.</h1>");
+    } else {
+        response.getWriter().println("<h1>Hello " + name + "!</h1>");
+    }
+
+    
+
```
Comment (HiramSilvey) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Delete unnecessary whitespace.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,7 +26,16 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, I don't know what your name is.</h1>");
+    } else {
+        response.getWriter().println("<h1>Hello " + name + "!</h1>");
```
Comment (HiramSilvey) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Even though there aren't any obvious dangers of exposing this site to a reflective XSS attack, I still think it's probably better to avoid it here if possible. Could you add a quick check before using name to ensure it only contains alphanumeric characters? It looks like this can be accomplished using regex, as seen here: https://www.thecrazyprogrammer.com/2015/07/how-to-check-string-is-alphanumeric-in-java.html

**Review by HiramSilvey — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (keiffer01) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Yeah it's probably good practice to make it safe anyways. I went and added some sanitization using regex as you suggested.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -26,7 +26,15 @@
...
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, looks like you didn't give anything.</h1>");
+    } else if (!name.matches("^[A-Za-z0-9 ]*$")){
+        response.getWriter().println("<h1>Sorry, I only accept letters, numbers, and spaces.</h1>");
+    } else {
+        response.getWriter().println("<h1>Hello " + name + "!</h1>");
```
Comment (beiebi0) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: indents.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -26,7 +26,15 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, looks like you didn't give anything.</h1>");
+    } else if (!name.matches("^[A-Za-z0-9 ]*$")){
```
Comment (beiebi0) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Please create a const static variable for "^[A-Za-z0-9 ]*$".

**Review by beiebi0 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (keiffer01) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Okay, I made and pushed the changes you suggested.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,9 +24,23 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  // Regex that accepts letters, numbers, and spaces
+  private static final String regex = "^[A-Za-z0-9 ]*$";
```
Comment (HiramSilvey) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Could you rename this to something more descriptive of what the regex represents? Additionally, Google's Java style requires constants are in UPPER_SNAKE_CASE. Example solutions: "ALPHANUMERIC" or "ALPHANUMERIC_WITH_SPACES".
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

**Review by HiramSilvey — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (keiffer01) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Oh I didn't know that there was a consolidated Java style guide for Google, thanks for the link! I've renamed the const variable.

**Review by HiramSilvey — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -24,9 +24,23 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+    throws IOException {
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println(
+          "<h1>Oops, looks like you didn't give anything.</h1>");
+    } else if (!name.matches(ALPHANUMERIC_WITH_SPACES)){
```
Comment (beiebi0) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: spaces needed 
```suggestion
    } else if (!name.matches(ALPHANUMERIC_WITH_SPACES)) {
```

**Review by beiebi0 — CHANGES_REQUESTED** [review]
Will approve once you fix the space. Almost there!

**Review by beiebi0 — APPROVED** [review]
